### DISCUSSION TITLE
Like/Approve and Dislike/Disapprove Training for the AI

### DISCUSSION THREAD

**Discussion by lonejohnwolf** [lllyasviel/Fooocus#2721] [Ideas]
I'd absolutely LOVE if there were to be a system implemented which allows users to either like, dislike, or neither on an image after it has been made, as a form of training for the image generator. If it can get real time feedback and connect patterns on which images are bad or good for the same prompt, then it can actively improve over time. I'm not sure how this would work or get set up, but it'd be so amazing to see! 

If it is at all possible, I'd love for a system like this to be added in, even if it's only client-side. I will HAPPILY create hundreds of images in a single day, then go over each and every one, liking the ones that were good and followed the prompt well, leaving the ones that weren't good but still followed the prompt neutral, and disliking the ones that failed to follow the prompt. 

I believe that something like this would greatly improve the image generation over time. And if the data isn't directly inserted into the AI on the website automatically, maybe people can share their training data somehow and insert that after it gets confirmed to be legit. 

This is my idea, and I genuinely think this is one of the best possible things to add to an image generator!

**Comment by mashb1t** [comment]
Rating generated images will not improve the output as long as the base model used to create the image isn't manually fine-tuned.
Training is not part of Fooocus' responsibilities or target features, as there are tools designed for this specific purpose with much more detail and functionality, such as frameworks for [RLHF](https://huggingface.co/blog/rlhf) and kohya_ss for training. As Fooocus is not a cloud service but a self-hosted platform, there is no centralized entity to monitor and improve on the base models with RLHF.
You can nevertheless save the good images and define a timeframe after which you then start with the fine-tuning of the base model or create a LoRA based on these generated images.
Just a reminder: this process will take a while and needs beefy hardware.

You can certainly create a component, integrare it and save ratings for yourself, but the majority of users will not train models themselves.

**Reply by lonejohnwolf** [reply]
Well I would want a feature that functions like how I described at the very least to work locally on my own computer, as there's plenty of things that Fooocus doesn't understand how to make images of, and I don't see any possible way to train it on what any of these things look like as a guy who just wants to get various specific images but doesn't know even the first thing about programming an AI. 

I really hope that Fooocus gets improvements with understanding what things are supposed to look like from the main source of training or wherever it originated from, because so far I just think it's a pretty great image generator, but nothing too amazing yet.

**Reply by mashb1t** [reply]
@lonejohnwolf You can only train models/loras etc. based on your input, not Fooocus, as Fooocus is based on these. What you can do is to use image prompts and references to up to 4 images you like+ mix with inpaint for improved control, but no RLHF.
